app/src/main/kotlin/com/mapbox/vision/teaser/MainActivity.kt

```diff
@@ -2,30 +2,58 @@ package com.mapbox.vision.teaser
 
 import android.content.Intent
 import android.view.View
-import android.widget.ImageView
 import android.widget.LinearLayout
+import android.widget.Toast
 import com.mapbox.vision.VisionManager
+import com.mapbox.vision.VisionReplayManager
+import com.mapbox.vision.mobile.core.models.FrameStatistics
 import com.mapbox.vision.teaser.view.BaseTeaserActivity
 import com.mapbox.vision.teaser.view.show
 import com.mapbox.vision.safety.VisionSafetyManager
+import com.mapbox.vision.teaser.MainActivity.VisionManagerMode.*
 import com.mapbox.vision.teaser.ar.ArMapActivity
+import com.mapbox.vision.teaser.replayer.ArReplayNavigationActivity
+import com.mapbox.vision.teaser.replayer.ReplayModeFragment
+import com.mapbox.vision.teaser.view.hide
 import com.mapbox.vision.view.VisionView
+import kotlinx.android.synthetic.main.activity_main.*
 
-class MainActivity : BaseTeaserActivity() {
+class MainActivity : BaseTeaserActivity(), ReplayModeFragment.OnClickModeItemListener {
+
+    enum class VisionManagerMode {
+        Realtime,
+        Replay,
+        Recorder
+    }
+
+    private var mode = Realtime
+    private var sessionPath = ""
 
     override fun initViews(root: View) {
         root.findViewById<LinearLayout>(R.id.ar_navigation_button_container).apply {
-            setOnClickListener { startActivity(Intent(this@MainActivity, ArMapActivity::class.java)) }
+            setOnClickListener {
+                when (mode) {
+                    Realtime -> startActivity(Intent(this@MainActivity, ArMapActivity::class.java))
+                    Replay -> startArSession()
+                    else -> return@setOnClickListener
```

Comment:
nit: for enum better 
```suggestion
                    Recorder -> return@setOnClickListener
```

Reply:
fixed here, but removed Recorder option in next PRs